bin/node/runtime/src/lib.rs

```diff
@@ -845,6 +869,16 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl sp_session::SessionMembership<Block> for Runtime {
+		fn generate_session_membership_proof(
+			session_key: (sp_core::crypto::KeyTypeId, Vec<u8>),
```

Comment:
Shouldn't the `Vec<u8>` here be a concrete key type?

Reply:
The session module only deals with opaque key types.

Reply:
That is true, but the implementation of `KeyOwnerProofSystem` is written as

```rust
impl<T: Trait, D: AsRef<[u8]>> frame_support::traits::KeyOwnerProofSystem<(KeyTypeId, D)>
	for Module<T>
```

I did it this way so that modules can avoid using `Vec`ly-typed keys with the `KeyOwnerProofSystem`, because the underlying bytes are the same. It's more understandable and more efficient to avoid `Vec`.

Reply:
https://github.com/paritytech/polkadot/pull/951

Reply:
Updated to do the same thing.